src/peft/utils/other.py

```diff
@@ -63,18 +63,26 @@ def starcoder_model_postprocess_past_key_value(past_key_values):
     return tuple(result)
 
 
-def prepare_model_for_kbit_training(model, use_gradient_checkpointing=True):
+def prepare_model_for_kbit_training(model, use_gradient_checkpointing=True, gradient_checkpointing_kwargs=None):
```

Comment:
I just now notice that we should probably mention that this function only works on transformers (-compatible) models, passing a custom model won't work here.

Reply:
Yes makes sense, updated the docstring accordingly